### ISSUE TITLE
fft on AFArray sub-array error

### ISSUE THREAD

**Issue by AshtonSBradley** [JuliaComputing/ArrayFire.jl#136]
Hi, I want to fft a sub-array of a 3D array, on the GPU. To fft over dimensions 1 and 2 of a 3D array, an example of syntax that works for Base.fft is:
```julia
m=2048;
k=10;
a = complex(randn(Float32, m,m,k),randn(Float32,m,m,k))::Array{Complex{Float32},3};
P = plan_fft!(a,[1,2])
b = P*a;
```
If I try to call fft! for an AFArray (which I assume implements the same behaviour, without planning, according to the documentation), then

```julia
ad=AFArray(a);
julia> fft!(ad,[1,2]);
ERROR: MethodError: no method matching af_fft3_inplace(::ArrayFire.AFArray{Complex{Float32},3}, ::Array{Int64,1})
Closest candidates are:
  af_fft3_inplace(::ArrayFire.AFArray{T,N}, ::Float64) at /home/abradley/.julia/v0.5/ArrayFire/src/wrap.jl:1823
 in fft!(::ArrayFire.AFArray{Complex{Float32},3}, ::Array{Int64,1}) at /home/abradley/.julia/v0.5/ArrayFire/src/signal.jl:143
```

Am I making some syntax mistake here, or should the base behaviour carry over?

**Comment by bradleyashton** [comment]
This seems a bit odd, as the documentation suggests that fft! should be fully supported... does this mean that if I want to do batch `fft` of many 2D arrays, I shouldn't be stacking them in a 3D array?

**Comment by pavanky** [comment]
@bradleyashton What does `[1, 2]` signify here? I don't think `ArrayFire.jl` supports that parameter.

**Comment by bradleyashton** [comment]
In the CPU code, it means "only do the transform over dimensions 1 and 2 of the array".

**Comment by pavanky** [comment]
@bradleyashton Ah that makes sense. It looks like those options are not enabled for the wrapper and because of that it can not distinguish between performing 3D FFT on 3D data as opposed to batched 2D FFT.

@ranjanan would it be possible to add that to the wrapper?

**Comment by jgbos** [comment]
Curious if this being worked on?  In addition to adding the capability to do a batch of 2D FFTs, it appears that we cannot do an FFT along any dimension beside 1 (line 90 and 100 in `signal.jl`)

**Comment by ranjanan** [comment]
@jgbos I think that might possibly be a limitation of the underlying C++ library.

**Comment by jgbos** [comment]
So it looks like for my specific applications this is possible with `af_fft2`.  It will automatically batch many 2D FFTs given an array larger then a matrix, but the Julia implementation only calls `af_fft2` for matrix types.  Don't know if this means you should add an `fft2` function or provide a `region` input to support julia `fft` interface.

**Comment by gaika** [comment]
fixed in #121 - there is `fft2` that would do 2D FFT on a 3D matrix in batch mode. Also `fft1`, `ifft2`, etc.
